src/serialization/compact/DefaultCompactReader.ts

```diff
@@ -0,0 +1,1284 @@
+        const fieldKind = fd.kind;
+        switch (fieldKind) {
+            case FieldKind.BOOLEAN:
+                return this.getBooleanWithFieldDescriptor(fd);
+            case FieldKind.NULLABLE_BOOLEAN:
+                return this.getVariableSizeAsNonNull(fd, reader => reader.readBoolean(), 'Boolean');
+            default:
+                throw DefaultCompactReader.toFieldKindIsNotOneOfError(
+                    fieldKind, fieldName, [FieldKind.BOOLEAN, FieldKind.NULLABLE_BOOLEAN]
+                );
+        }
+    }
+
+    private getVariableSizeAsNonNull<T>(
+        fieldDescriptor: FieldDescriptor, readFn: (reader: ObjectDataInput) => T, methodSuffix: string): T {
+        const value = this.getVariableSize(fieldDescriptor, readFn);
+        if (value === null) {
+            throw CompactExceptions.toExceptionForUnexpectedNullValue(fieldDescriptor.fieldName, methodSuffix);
+        }
+        return value;
+    }
+
+    private getBooleanWithFieldDescriptor(fieldDescriptor: FieldDescriptor): boolean {
+        const booleanOffset = fieldDescriptor.offset;
+        const bitOffset = fieldDescriptor.bitOffset;
+        const getOffset = booleanOffset + this.dataStartPosition;
+        const lastByte = this.input.readByte(getOffset);
+        return ((lastByte >>> bitOffset) & 1) !== 0;
+    }
+
+    getInt8(fieldName: string): number {
+        const fd = this.getFieldDefinition(fieldName);
+        const fieldKind = fd.kind;
+        switch (fieldKind) {
+            case FieldKind.INT8:
+                return this.input.readInt8(this.readFixedSizePosition(fd));
+            case FieldKind.NULLABLE_INT8:
+                return this.getVariableSizeAsNonNull(fd, reader => reader.readInt8(), 'Byte');
+            default:
+                throw DefaultCompactReader.toFieldKindIsNotOneOfError(
+                    fieldKind, fieldName, [FieldKind.INT8, FieldKind.NULLABLE_INT8]
+                );
+        }
+    }
+
+    getDate(fieldName: string): LocalDate {
+        return this.getVariableSizeByNameAndKind(fieldName, FieldKind.DATE, IOUtil.readLocalDate);
+    }
+
+    getDecimal(fieldName: string): BigDecimal {
+        return this.getVariableSizeByNameAndKind(fieldName, FieldKind.DECIMAL, IOUtil.readDecimal);
+    }
+
+    getFloat64(fieldName: string): number {
+        const fd = this.getFieldDefinition(fieldName);
+        const fieldKind = fd.kind;
+        switch (fieldKind) {
+            case FieldKind.FLOAT64:
+                return this.input.readDouble(this.readFixedSizePosition(fd));
+            case FieldKind.NULLABLE_FLOAT64:
+                return this.getVariableSizeAsNonNull(fd, reader => reader.readDouble(), 'Double');
+            default:
+                throw DefaultCompactReader.toFieldKindIsNotOneOfError(
+                    fieldKind, fieldName, [FieldKind.FLOAT64, FieldKind.NULLABLE_FLOAT64]
+                );
+        }
+    }
+
+    getFieldKind(fieldName: string): FieldKind {
```

Comment:
Once the defaultcompactreader was a generic record implementor but then I removed it probably because users won't directly get the class instance. However, typescript or eslint does not show any unused symbol warnings and that is strange.